### ISSUE TITLE
if statements clobber @_ in Rakudo

### ISSUE THREAD

**Issue by p6rt** [Raku/old-issue-tracker#1488]
Migrated from [rt.perl.org#72600](https://rt-archive.perl.org/perl6/Ticket/Display.html?id=72600) (status was 'resolved')

Searchable as RT72600$

**Comment by niner** [comment]
I just removed that spec test. As I wrote in the commit message:

Arguably the previous unexpected behavior is totally correct.
    
The block in if True { } is a real block, not just a syntactic grouping. That's
how it's advertised as "strangely consistent" and why you can give it a
signature with if True -> $true { }
    
@_ is a magical auto-declaring variable that contains the surrounding block's
positional arguments. Since the if statement's block is such a block, that's
what it should be declared on.
  
m: sub foo { dd @_; if True -> $true { dd @_ } }(1, 2)
rakudo-moar 0730199b3: OUTPUT: «===SORRY!=== Error while compiling
<tmp>␤Placeholder variable '@_' cannot override existing signature␤at
<tmp>:1␤------> sub foo { dd @_; if True <HERE>-> $true { dd @_ } }(1, 2)␤»
    
After all it does complain about that if block having a signature in this example
    
But wrap it in another sub:
m: sub bar(*@_) { sub foo { dd @_; if True { dd @_ } }(1, 2) }()
rakudo-moar 0730199b3: OUTPUT: «Too many positionals passed; expected 0
arguments but got 2␤  in sub foo at <tmp> line 1␤  in sub bar at <tmp> line 1␤
in block <unit> at <tmp> line 1␤␤»

I think all the existing behavior including the incidental pasisng of the test just
comes from a check somewhere whether @_ is visible lexically or not with no
regards of where it's actually declared.

In short, @_ *should* implicitly be declared on that if block. That this
can be surprising to users is just a side effect of implicitly declaring
such a variable in the first place.
